android/src/app/organicmaps/MwmJobIntentService.java

```diff
@@ -8,32 +9,41 @@
 import app.organicmaps.util.CrashlyticsUtils;
 import app.organicmaps.util.log.Logger;
 
-@SuppressWarnings("deprecation") // https://github.com/organicmaps/organicmaps/issues/4531
-public abstract class MwmJobIntentService extends JobIntentService
-{
-  private static final String TAG = MwmJobIntentService.class.getSimpleName();
-
-  @NonNull
-  protected String getTag()
-  {
-    return getClass().getSimpleName();
-  }
-
-  protected abstract void onHandleWorkInitialized(@NonNull Intent intent);
-
-  @Override
-  protected void onHandleWork(@NonNull Intent intent)
-  {
-    MwmApplication app = MwmApplication.from(this);
-    String msg = "onHandleWork: " + intent;
-    Logger.i(TAG, msg);
-    CrashlyticsUtils.INSTANCE.log(Log.INFO, getTag(), msg);
-    if (!app.arePlatformAndCoreInitialized())
-    {
-      Logger.w(TAG, "Application is not initialized, ignoring " + intent);
-      return;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import java.util.logging.Logger;
+
+public abstract class MwmJobIntentService extends Worker {
+    private static final String TAG = MwmJobIntentService.class.getSimpleName();
+
+    @NonNull
+    protected String getTag() {
+        return getClass().getSimpleName();
     }
 
-    onHandleWorkInitialized(intent);
-  }
+    protected abstract void onHandleWorkInitialized(@NonNull Intent intent);
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        Context applicationContext = getApplicationContext();
```

Comment:
All non-modified vars should have `final` keyword for better readability.

Reply:
I did not test it.
i'm sorry for wasting your time.

Reply:
Please test it by adding additional logs, and by trying to edit some features in OSM (but you need to build a beta build for that, not a debug one).

Reply:
ok, I'll do that.
Thanks for guidance